internal/services/netapp/netapp_volume_group_resource.go

```diff
@@ -0,0 +1,642 @@
+)
+
+type NetAppVolumeGroupResource struct{}
+
+type NetAppVolumeGroupModel struct {
+	Name                  string                    `tfschema:"name"`
+	ResourceGroupName     string                    `tfschema:"resource_group_name"`
+	Location              string                    `tfschema:"location"`
+	AccountName           string                    `tfschema:"account_name"`
+	GroupDescription      string                    `tfschema:"group_description"`
+	ApplicationType       string                    `tfschema:"application_type"`
+	ApplicationIdentifier string                    `tfschema:"application_identifier"`
+	Volumes               []NetAppVolumeGroupVolume `tfschema:"volume"`
+}
+
+var _ sdk.Resource = NetAppVolumeGroupResource{}
+
+func (r NetAppVolumeGroupResource) ModelObject() interface{} {
+	return &NetAppVolumeGroupModel{}
+}
+
+func (r NetAppVolumeGroupResource) ResourceType() string {
+	return "azurerm_netapp_volume_group"
+}
+
+func (r NetAppVolumeGroupResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return volumegroups.ValidateVolumeGroupID
+}
+
+func (r NetAppVolumeGroupResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"account_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: netAppValidate.AccountName,
+		},
+
+		"group_description": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+
+		"application_type": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringInSlice([]string{
+				string(volumegroups.ApplicationTypeSAPNegativeHANA),
+			}, false),
+		},
+
+		"application_identifier": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringLenBetween(1, 3),
+		},
+
+		"volume": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MinItems: 2,
+			MaxItems: 5,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"id": {
+						Type:     pluginsdk.TypeString,
+						Computed: true,
+					},
+
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: netAppValidate.VolumeName,
+					},
+
+					"capacity_pool_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: azure.ValidateResourceID,
+					},
+
+					"proximity_placement_group_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: azure.ValidateResourceID,
+					},
+
+					"volume_spec_name": {
```

Comment:
With the upcoming update on AVG feature for another database type, the volume spec types will increase to additional eleven types and if I recall we might be able to have multiple of those and it will get harder and harder to create volume types each time ANF introduces a new DB type.

For context, this feature is very DB centric, it is starting with support for SAP-HANA but will grow to other databases over time, with the second coming in the next few months.

Reply:
Ohhhh that makes sense . Thank you! I'm good with keeping it as is then